common/src/proto.rs

```diff
@@ -34,6 +34,7 @@ pub struct StateDelta {
     /// Highest input generation received prior to `step`
     pub latest_input: u16,
     pub positions: Vec<(EntityId, Position)>,
+    pub character_velocities: Vec<(EntityId, na::Vector3<f32>)>,
     pub character_orientations: Vec<(EntityId, na::UnitQuaternion<f32>)>,
```

Comment:
Since these two fields always cover exactly the same entities, it'd be cleaner/lighter/more convenient to fold them into a single `Vec<(EntityId, CharacterState)>`.

Reply:
That makes sense. Would it also make sense to use `Character` instead of a new struct `CharacterState`, and have the "name" live in its own component? This seems more inline with ECS, and it would allow `Character` to be used for something like an NPC.

Reply:
The existing `Character` struct should remain, and should have a `CharacterState` member.

> This seems more inline with ECS

The network data model will never and should never exactly match the in-memory data model. Different components of the system have different concerns. For example, the network layer should only send updates for data that is subject to change, but static and dynamic data that are always associated with each other should not be needlessly separated in the ECS.